Cesium3DTilesSelection/src/WebMapServiceRasterOverlay.cpp

```diff
@@ -0,0 +1,410 @@
+      const CesiumGeometry::QuadtreeTilingScheme& tilingScheme,
+      const CesiumGeometry::Rectangle& coverageRectangle,
+      const std::string& url,
+      const std::vector<IAssetAccessor::THeader>& headers,
+			const std::string& version,
+      const std::string& layername,
+      const std::string& fileExtension,
+      uint32_t width,
+      uint32_t height,
+      uint32_t minimumLevel,
+      uint32_t maximumLevel)
+      : QuadtreeRasterOverlayTileProvider(
+            owner,
+            asyncSystem,
+            pAssetAccessor,
+            credit,
+            pPrepareRendererResources,
+            pLogger,
+            projection,
+            tilingScheme,
+            coverageRectangle,
+            minimumLevel,
+            maximumLevel,
+            width,
+            height),
+        _url(url),
+        _headers(headers),
+        _version(version),
+        _layername(layername),
+        _fileExtension(fileExtension) {}
+
+  virtual ~WebMapServiceTileProvider() {}
+
+protected:
+  virtual CesiumAsync::Future<LoadedRasterOverlayImage> loadQuadtreeTileImage(
+      const CesiumGeometry::QuadtreeTileID& tileID) const override {
+ 
+    LoadTileImageFromUrlOptions options;
+    options.rectangle = this->getTilingScheme().tileToRectangle(tileID);
+    options.moreDetailAvailable = tileID.level < this->getMaximumLevel();
+
+    //SPDLOG_LOGGER_INFO(getLogger(), "tileID: level>" + std::to_string(tileID.level) + "; x>" + std::to_string(tileID.x) + "; y>" + std::to_string(tileID.y));
+
+    const CesiumGeospatial::GlobeRectangle tileRectangle =
+        CesiumGeospatial::unprojectRectangleSimple(this->getProjection(), options.rectangle);
+
+    const std::string urlTemplate =
+        "{baseUrl}?request=GetMap&TRANSPARENT=TRUE&version={version}&service=wms&"
+        "format=image.png&styles="
+        "&width={width}&height={height}&bbox={minx},{miny},{maxx},{maxy}"
+        "&layers={layername}&crs=EPSG:4326";
+
+    const auto radiansToDegrees = [](double rad) { 
+        return std::to_string(CesiumUtility::Math::radiansToDegrees(rad));
+    };
+
+    const std::map<std::string, std::string> urlTemplateMap = { 
+      {"baseUrl", this->_url},
+      {"version", this->_version},
+      {"maxx", radiansToDegrees(tileRectangle.getNorth())},
+      {"maxy", radiansToDegrees(tileRectangle.getEast())},
+      {"minx", radiansToDegrees(tileRectangle.getSouth())},
+      {"miny", radiansToDegrees(tileRectangle.getWest())},
+      {"layername", this->_layername},
+      {"width", std::to_string(this->getWidth())},
+      {"height", std::to_string(this->getHeight())}
+    };
+
+    std::string url = CesiumUtility::Uri::substituteTemplateParameters(
+        urlTemplate,
+        [
+          map=&urlTemplateMap
+        ](const std::string& placeholder) {
+          return map->count(placeholder) == 0 ? "{" + placeholder + "}"
+                                              : map->at(placeholder);
```

Comment:
It's a little clearer and more performant (not that the performance improvement really matters here) to use iterators, e.g.:

```suggestion
          auto it = map.find(placeholder);
          return it == map.end() ? "{" + placeholder + "}" : it->second;
```